src/model/AbstractRepository.ts

```diff
@@ -1,13 +1,18 @@
-import { mapping } from "cassandra-driver";
+import { AbstractEntity } from "./AbstractEntity";
+import { mapper } from "./CassandraClient";
 
-export abstract class AbstractRepository<T extends { [key: string]: any }, U> {
+export abstract class AbstractRepository<T, U extends AbstractEntity> {
   public abstract get tableName(): string;
   public abstract get entityName(): string;
-  public abstract get mapper(): mapping.ModelMapper<U>;
 
-  protected abstract convertEntityToDTO(row: U): T;
```

Comment:
Another suggestion to better respect the Repository pattern is to remove the mapping (i.e. convertEntityToDto and vice a versa) and put it inside the entity model class itself, therefore avoiding the antipattern of anemic objects and allowing the encapsulating of the conversion login inside entities and DTOs. And makes easier to unit test them.

Converting entities is not usually part of the Abstract Repository pattern.

Reply:
Instead of a separate "entity" interface, I am now using the driver's Mapping function to parse the objects into DTOs